python/desc/imsim/imSim.py

```diff
@@ -198,105 +206,5 @@ def parsePhoSimInstanceFile(fileName, numRows):
-    # specified.
-    numRows = arguments.numrows
-    print "Reading", numRows, "rows from the instance file."
-
-    # Start by reading a PhoSim instance file.
-    logger.info('Reading PhoSim instance catalog')
-
-    # The PhoSim instance file contains both pointing commands and objects.  The
-    # parser will split them and return a both phosim command dictionary and a
-    # dataframe of objects.
-    commandDictionary, phoSimDataBase = parsePhoSimInstanceFile(instanceFile.name, numRows)
-
-    # Now further sub-divide the source dataframe into stars and galaxies.
-    starDataBase = phoSimObjectList.query("galSimType == 'pointSource'")
-    galaxyDataBase = phoSimObjectList.query("galSimType == 'sersic'")
-
-    # Access the relevant pointing commands from the dictionary.
-    visitIDString = str(int(commandDictionary['obshistid'][0]))
-    mjd = commandDictionary['mjd'][0]
-    rightAscension = commandDictionary['rightascension'][0]
-    declination = commandDictionary['declination'][0]
-    rotSkyPosition = commandDictionary['rotskypos'][0]
-
-    # Now build the ObservationMetaData with values taken from the PhoSim
-    # commands at the top of the instance file.
-
-    # We need to set the M5 limiting magnitudes etc.
-    defaults = LSSTdefaults()
-
-    # defaults.seeing('r') doesn't exist in my version of the stack yet.
-    # Manually set to 0.7
-    obs = ObservationMetaData(pointingRA=rightAscension, pointingDec=declination,
-                              mjd=mjd, rotSkyPos=rotSkyPosition,
-                              bandpassName=['r'],
-                              m5=[defaults.m5('r')],
-                              seeing=[0.7])
-
-    # Simulate the objects in the Pandas Dataframes. I monkey patched the
-    # abstract base class GalSimBase to take my dataFrame instead of using the
-    # internal databases. This is a hack until we rethink the class design.
-    #
-    # Additionally, I have set the PSF and noise etc globally for all of the
-    # derived classes in the __init__.  For simulation the LSST this should be
-    # the same between all types of objects.
-
-    # First simulate stars
-    phoSimStarCatalog = GalSimStars(starDataBase, obs)
-    phoSimStarCatalog.camera = LsstSimMapper().camera
-    phoSimStarCatalog.get_fitsFiles()
-
-    # Now galaxies
-    phoSimGalaxyCatalog = GalSimGalaxies(galaxyDataBase, obs)
-    phoSimGalaxyCatalog.copyGalSimInterpreter(phoSimStarCatalog)
-    phoSimGalaxyCatalog.get_fitsFiles()
-
-    # Write out the fits files
-    output_dir = arguments.outdir
-    if not os.path.isdir(output_dir):
-        os.makedirs(output_dir)
-    phoSimGalaxyCatalog.write_images(nameRoot=os.path.join(output_dir, 'e-image_')+visitIDString)
-
-if __name__ == "__main__":
-    main(sys.argv)
+    return PhoSimInstanceCatalogContents(commandDictionary, phoSimObjectList,
+                                         phoSimHeaderCards, phoSimSources)
```

Comment:
The other two items were `global` in the original code so that the caller could have access. In order to avoid that, I made them part of the returning tuple so that they were still available.